streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -452,21 +336,12 @@ public Long apply(K1 aggKey, V value, Long aggregate) {
         // read the intermediate topic
         topology.addSource(sourceName, keySerde.deserializer(), changedValueDeserializer, topic);
 
-        // aggregate the values with the aggregator and local store
-        topology.addProcessor(reduceName, aggregateSupplier, sourceName);
-        topology.addStateStore(aggregateStore, reduceName);
-
-        // return the KTable representation with the intermediate topic as the sources
-        return new KTableImpl<>(topology, reduceName, aggregateSupplier, Collections.singleton(sourceName));
+        return new KGroupedTableImpl<>(topology, sourceName, sourceName, keySerde, valueSerde);
```

Comment:
If we can defer the creation of the repartition sink node and source node to KGroupedTableImpl, then we do not need the name in `groupBy` as it can just use the name of the resulted aggregated `KTable`.